src/UKHO.ADDS.EFS.Builder.S100/Pipelines/Assemble/ProductSearchNode.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System.Text;
+using System.Web;
+using Microsoft.Extensions.Options;
+using UKHO.ADDS.Clients.FileShareService.ReadOnly;
+using UKHO.ADDS.Clients.FileShareService.ReadOnly.Models;
+using UKHO.ADDS.EFS.Builder.S100.Pipelines.Assemble.Logging;
+using UKHO.ADDS.EFS.Builder.S100.Pipelines.Assemble.Models;
+using UKHO.ADDS.EFS.Configuration.Builder;
+using UKHO.ADDS.EFS.Exceptions;
+using UKHO.ADDS.Infrastructure.Pipelines;
+using UKHO.ADDS.Infrastructure.Pipelines.Nodes;
+
+namespace UKHO.ADDS.EFS.Builder.S100.Pipelines.Assemble
+{
+    public class ProductSearchNode : ExchangeSetPipelineNode
+    {
+        private readonly FileShareServiceConfiguration _fileShareServiceConfiguration;
```

Comment:
Please use IOptions here and we can use IConfiguration in Program.cs to map the configuration.
```suggestion
        private readonly IOptions<FileShareServiceConfiguration> _fileShareServiceConfig;
```